code/modules/mob/living/simple_animal/hostile/statue.dm

```diff
@@ -60,137 +63,95 @@
 	if(creator)
 		src.creator = creator
 
-/mob/living/simple_animal/hostile/statue/Move(var/turf/NewLoc)
+/mob/living/simple_animal/hostile/statue/Move(turf/NewLoc)
 	if(can_be_seen(NewLoc))
-		icon_state = "angelseen"
 		if(client)
 			src << "<span class='warning'>You cannot move, there are eyes on you!</span>"
 		return 0
-	icon_state = "angel"
 	return ..()
 
-/mob/living/simple_animal/hostile/statue/process_ai()
-	. = ..()
-	if(!.)
-		return
-	spawn() //for saftey
-		checkTargets()
 
-/mob/living/simple_animal/hostile/statue/proc/checkTargets()
-	if(target)
+/mob/living/simple_animal/hostile/statue/process_ai()
+	..()
+	if(!client && target) // If we have a target and we're AI controlled
 		var/mob/watching = can_be_seen()
+		// If they're not our target
 		if(watching && watching != target)
+			// This one is closer.
 			if(get_dist(watching, src) > get_dist(target, src))
 				LoseTarget()
 				GiveTarget(watching)
 
 /mob/living/simple_animal/hostile/statue/AttackingTarget()
-	if(!can_be_seen())
-		icon_state = "angelattack"
+	if(can_be_seen(get_turf(loc)))
+		if(client)
+			src << "<span class='warning'>You cannot attack, there are eyes on you!</span>"
+			return
+	else
 		..()
 
 /mob/living/simple_animal/hostile/statue/DestroySurroundings()
-	if(!can_be_seen())
+	if(!can_be_seen(get_turf(loc)))
 		..()
 
 /mob/living/simple_animal/hostile/statue/face_atom()
-	if(!can_be_seen())
+	if(!can_be_seen(get_turf(loc)))
 		..()
 
-/mob/living/simple_animal/hostile/statue/UnarmedAttack()
-	if(can_be_seen())
-		icon_state = "angelseen"
-		if(client)
-			src << "<span class='warning'>You cannot attack, there are eyes on you!</span>"
-		return
-	icon_state = "angelattack"
-	..()
-
-/mob/living/simple_animal/hostile/statue/proc/can_be_seen(var/turf/destination)
+/mob/living/simple_animal/hostile/statue/proc/can_be_seen(turf/destination)
+	if(!cannot_be_seen)
+		return null
 	// Check for darkness
 	var/turf/T = get_turf(loc)
-
-	var/light_amount
-	var/DLlight_amount
-
-	if(T.dynamic_lighting)
-		light_amount = T.get_lumcount()
-	else
-		light_amount = 5
-
-	if(destination && destination.dynamic_lighting)
-		DLlight_amount = destination.get_lumcount()
-	else
-		DLlight_amount = 5
-
-	if(T && destination)
-		//Don't check it twice if our destination is the tile we are on or we can't even get to our destination
+	if(T && destination && T.lighting_overlay)
+		if(T.get_lumcount() * 10 < 1 && destination.get_lumcount() * 10 < 1) // No one can see us in the darkness, right?
```

Comment:
was largely following the standard practice elsewhere, where `get_lumcount` is utilized.